zkevm-circuits/src/bytecode_circuit.rs

```diff
@@ -4,7 +4,9 @@
 pub mod bytecode_unroller;
 /// Bytecode circuit
 pub mod circuit;
-/// Bytecode circuit tester
-#[cfg(any(feature = "test", test))]
-pub mod dev;
 pub(crate) mod param;
+/// Bytecode circuit tester
+#[cfg(any(feature = "test", test, feature = "test-circuits"))]
```

Comment:
Make sense.

Reply:
I took a look at the feature selection and recognized that it will be not easily possible to keep a distinction with `mod test `(feature=test) and only the `test-circuits` feature that only requires the circuit implementation to be exported. Basically we have to add `cfg(any(feature = "test", test))]` and `cfg(any(feature = "test", test, feature = "test-circuits"))]` selectively in new places for each test.rs file. I think that is cumbersome and to not make this further complicated I opt into removing the `test-circuits` feature and replace it with something else later. What do you think?

cc/ @ed255 Might also want to chime in

Reply:
> I took a look at the feature selection and recognized that it will be not easily possible to keep a distinction with `mod test `(feature=test) and only the `test-circuits` feature that only requires the circuit implementation to be exported. Basically we have to add `cfg(any(feature = "test", test))]` and `cfg(any(feature = "test", test, feature = "test-circuits"))]` selectively in new places for each test.rs file. 

hi, @pinkiebell  , after viewing the #1144 . I've met the same problem.
I've been sperate the `iml Circuit` part as a dev mod to have a distinct board between 'test-circuits' feature and `test` one.

> I think that is cumbersome and to not make this further complicated I opt into removing the `test-circuits` feature and replace it with something else later. What do you think?

It's better to kept the `test-circuits` features. Because the `impl circuit` part can be used in unit tests , integrated tests and benches. With test-circuits, integrated/benches or other testers can ignore the useless unit-test.